dclab/rtdc_dataset/writer.py

```diff
@@ -210,14 +216,33 @@ def store_feature(self, feat, data, shape=None):
                         + 'in the `info["feature shapes"]` key of '
                         + "your plugin feature.")
                     shape = data.shape[1:]
-            if shape == data.shape:
-                data = data.reshape(1, *shape)
-            elif shape == data.shape[1:]:
-                pass
+            # Condition for single or multiple events to be saved
+            if shape == data.shape or shape == data.shape[1:]:
+                # Condition for ragged/contour data (array of arrays)
+                if data.ndim == 1 and isinstance(data[0], np.ndarray):
+                    self.write_ragged(group=events, name=feat, data=data)
+                # Condition for single image or array of shape (H, W). If the
+                # given data is a 2D, it is more likely a single image feature
+                # therefore it will be saved with image attributes.
+                elif data.ndim == 2:
```

Comment:
Shouldn't you be checking against `shape` here and not `data.ndim`? The warning message above would otherwise be invalid. What if someone just stores a new scalar (temporary) feature? Then `data.ndim` is two but shape is `(1,)`.

Reply:
True, will change this.